proto/process/agent.proto

```diff
@@ -726,6 +735,7 @@ message Connections {
 	AgentConfiguration agentConfiguration = 7;
 	repeated string tags = 8;
 	map<string, int64> connTelemetryMap = 9;
+	repeated FailedConnection failedConns = 10;
```

Comment:
I think would be better to reuse Connection type here, to avoid confusion and only value previously set would be serialized in protobuf (not the default value)

s|	repeated FailedConnection failedConns = 10; | 	repeated Connection failedConns = 10; |

Reply:
all the field that can be set by the agent in failedConns would be useful

Reply:
It doesn't affect protobuf, but it does affect our in-memory size. I'd argue for a separate type here.